isis/tests/FunctionalTestsCsminit.cpp

```diff
@@ -0,0 +1,207 @@
+#include "csminit.h"
+#include "spiceinit.h"
+
+#include <QString>
+#include <iostream>
+
+#include "TestCsmPlugin.h"
+#include "Fixtures.h"
+#include "TestUtilities.h"
+#include "StringBlob.h"
+#include "FileName.h"
+
+#include <nlohmann/json.hpp>
+using json = nlohmann::json;
+
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/csminit.xml").expanded();
+
+class CSMPluginFixture : public TempTestingFiles {
+  protected:
+    const csm::Plugin *plugin;
+    Cube *testCube;
+    Pvl label;
+    QString isdPath;
+    QString filename;
+
+    void SetUp() override {
+      TempTestingFiles::SetUp();
+
+      // Create and populate default test ISD
+      json isd;
+      isd["name"] = "test_isd";
+      isd["test_param_one"] = "value_one";
+      isd["test_param_two"] = "value_two";
+
+        isdPath = tempDir.path() + "/default.json";
+        std::ofstream file(isdPath.toStdString());
+        file << isd;
+        file.flush();
+
+        std::ifstream cubeLabel("data/threeImageNetwork/cube1.pvl");
+        cubeLabel >> label;
+        testCube = new Cube();
+        filename = tempDir.path() + "/csminitCube.cub";
+        testCube->fromLabel(filename, label, "rw");
+        testCube->close();
+        
+        // There should only be one plugin, so get the first one
+        std::list<const csm::Plugin*> plugins = csm::Plugin::getList();
+        plugin = plugins.front();
+      }
+
+      void TearDown() override {
+        if (testCube) {
+          if (testCube->isOpen()) {
+            testCube->close();
+          }
+          delete testCube;
+          testCube = NULL;
+        }
+      }
+};
+
+TEST_F(CSMPluginFixture, CSMInitDefault) {
+  // Run csminit with defaults for everything besides FROM and ISD
+  QVector<QString> args = {
+    "from="+filename,
+    "isd="+isdPath
+  };
+
+  UserInterface options(APP_XML, args);
+  csminit(options);
+
+  testCube->open(filename);
+
+  // Get a model and a state string
+  StringBlob stateString("","CSMState");
+  testCube->read(stateString);
+
+  // Verify contents of the StringBlob's PVL label
+  PvlObject blobPvl = stateString.Label(); 
+  EXPECT_EQ(stateString.Name().toStdString(), "CSMState");
+  EXPECT_EQ(stateString.Type().toStdString(), "String"); 
+
+  // Check that the plugin can create a model from the state string
+  std::string modelName = QString(blobPvl.findKeyword("ModelName")).toStdString();
+  EXPECT_TRUE(plugin->canModelBeConstructedFromState(stateString.string(), modelName));
+
+  // Check blob label ModelName and Plugin Name 
+  EXPECT_EQ(QString(blobPvl.findKeyword("PluginName")).toStdString(), "TestCsmPlugin");
```

Comment:
Change the hard coded plugin name to `plugin->getPluginName()`

Reply:
It would be nice to get the model name from the model, but you would have to make a model object and I don't think that's worth it. Maybe you could make the model names public member variables of the models.